library/std/src/diagnostics.qs

```diff
@@ -217,5 +217,32 @@ namespace Microsoft.Quantum.Diagnostics {
         body intrinsic;
     }
 
-    export DumpMachine, DumpRegister, CheckZero, CheckAllZero, Fact, CheckOperationsAreEqual, StartCountingOperation, StopCountingOperation, StartCountingFunction, StopCountingFunction;
+    /// # Summary
+    /// Starts counting the number of qubits allocated. The counter is reset to zero.
+    ///
+    /// # Description
+    /// This operation allows you to count the number of qubits allocated until `StopCountingQubits` is called.
+    /// As part of starting the counting, the counter is reset to zero, which may override a previous count.
+    /// The counter is incremented only when a new unique qubit is allocated, so reusing the same qubit multiple times
+    /// across separate allocations does not increment the counter.
+    @Config(Unrestricted)
```

Comment:
Is this not interesting/useful for other profiles? For example, our BellState.qs sample uses only 2 qubits in 'unrestricted', but uses 8 if run under 'base profile'

Reply:
If this was allowed in Base profile you could never call it, because it returns a runtime dynamic integer, which is not allowed in Base profile.